spec/factories/storage_locations.rb

```diff
@@ -23,22 +23,24 @@
     organization { Organization.try(:first) || create(:organization) }
     square_footage { 100 }
     warehouse_type { StorageLocation::WAREHOUSE_TYPES.sample }
+    transient do
+      item_count { 1 }
+    end
 
     trait :with_items do
       transient do
-        item_count { 1 }
         item_quantity { 100 }
         item { nil }
       end
 
       after(:create) do |storage_location, evaluator|
-        if evaluator.item.nil?
+        if evaluator.item.nil? && evaluator.item_count != 0
```

Comment:
```suggestion
        if evaluator.item.nil? && !evaluator.item_count.zero?
```

Just some syntactic sugar